### PR TITLE
Update containers/storage and containers/image

### PR DESCRIPTION
**Pull request by nalind**
This bumps the versions of containers/storage and containers/image that we use, and updates the shortcuts that we take when committing to local storage to compensate for assumptions that it made which are no longer true (now it makes different ones).

### CODE REVIEW

**Comment by nalind** [comment]
[(13) Teach containers/image to take advantage of newer containers/storage features](https://trello.com/c/plhzr6ez/561-13-teach-containers-image-to-take-advantage-of-newer-containers-storage-features)

File: commit.go [hunk 1]
```diff
@@ -116,10 +130,10 @@ func (b *Builder) shallowCopy(dest types.ImageReference, src types.ImageReferenc
 		Size:   int64(len(config)),
 	}
 	_, err = destImage.PutBlob(bytes.NewReader(config), configBlobInfo)
-	if err != nil && len(config) > 0 {
+	if err != nil {
```
Comment (rhatdan) [inline commit.go]: Merge into a single line.

```
if _, err := destImage.PutBlob(bytes.NewReader(config), configBlobInfo); err != nil {
```

File: commit.go [hunk 2]
```diff
@@ -96,10 +85,35 @@ func (b *Builder) shallowCopy(dest types.ImageReference, src types.ImageReferenc
...
+		Compression: &compression,
+	}
+	layerDiff, err := b.store.Diff("", container.LayerID, diffOptions)
 	if err != nil {
-		return errors.Wrapf(err, "error writing dummy layer for image %q", transports.ImageName(dest))
+		return errors.Wrapf(err, "error reading layer %q from source image %q", container.LayerID, transports.ImageName(src))
+	}
+	defer layerDiff.Close()
+	// Write a copy of the layer as a blob, for the new image to reference.
+	_, err = destImage.PutBlob(layerDiff, types.BlobInfo{Digest: "", Size: -1})
```
Comment (rhatdan) [inline commit.go]: Single line if _, err := ...

File: commit.go [hunk 3]
```diff
@@ -96,10 +85,35 @@ func (b *Builder) shallowCopy(dest types.ImageReference, src types.ImageReferenc
...
+	}
+	// Extract the read-write layer's contents, using whatever compression the container image used to
+	// calculate the blob sum in the manifest.
+	switch compression {
+	case archive.Gzip:
+		logrus.Debugf("extracting layer %q with gzip", container.LayerID)
+	case archive.Bzip2:
+		// Until the image specs define a media type for bzip2-compressed layers, even if we know
+		// how to decompress them, we can't try to compress layers with bzip2.
+		return errors.New("media type for bzip2-compressed layers is not defined")
```
Comment (rhatdan) [inline commit.go]: Should we return ENOSUP?

File: commit.go (hunk 1 above)
Comment (nalind) [inline commit.go]: Fixing.

File: commit.go (hunk 2 above)
Comment (nalind) [inline commit.go]: Fixing.

**Comment by nalind** [comment]
Updated.

**Comment by rhatdan** [comment]
I hit the easy to understand stuff, have to rely on the Test suites for the rest. LGTM

**Comment by rhatdan** [comment]
@nalind needs rebase.

**Comment by rhatdan** [comment]
@nalind needs rebase.

**Comment by nalind** [comment]
@rhatdan, @TomSweeneyRedHat, is there anything you want changed here?

**Comment by TomSweeneyRedHat** [comment]
LGTM and I know it fixes up a number of older issues.  I'm also trusting in the tests to shake stuff out.  I'd say push the big red button.

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**